ludwig/utils/neuropod_utils.py

```diff
@@ -1,208 +1,119 @@
+import importlib.util
 import logging
 import os
-import shutil
+import tempfile
+from typing import Any, Dict, List
 
-import numpy as np
+import torch
 
-from ludwig import __file__ as ludwig_path
 from ludwig.api import LudwigModel
-from ludwig.backend import LOCAL_BACKEND, LocalBackend
-from ludwig.constants import (
-    BINARY,
-    CATEGORY,
-    NAME,
-    NUMBER,
-    PREDICTIONS,
-    PROBABILITIES,
-    PROBABILITY,
-    SEQUENCE,
-    SET,
-    TEXT,
-    TYPE,
-    VECTOR,
-)
-from ludwig.globals import MODEL_HYPERPARAMETERS_FILE_NAME, MODEL_WEIGHTS_FILE_NAME, TRAIN_SET_METADATA_FILE_NAME
-from ludwig.utils.data_utils import load_json
+from ludwig.constants import NAME
 
 logger = logging.getLogger(__name__)
 
 
-class LudwigNeuropodModelWrapper:
-    def __init__(self, data_root, backend):
-        self.ludwig_model = LudwigModel.load(data_root, backend=backend)
-
-    def __call__(self, **kwargs):
-        data_dict = kwargs
-        for key in data_dict:
-            data_dict[key] = np.squeeze(data_dict[key], axis=1)
-        predicted, _ = self.ludwig_model.predict(dataset=data_dict, return_type="dict")
-        # print(predicted, file=sys.stderr)
-        return postprocess_for_neuropod(predicted, self.ludwig_model.config)
-
+INFERENCE_MODULE_TEMPLATE = """
+from typing import Any, Dict, List, Union
+import torch
+
+class GeneratedInferenceModule(torch.nn.Module):
+    def __init__(self, inference_module):
+        super().__init__()
+        self.inference_module = inference_module
+
+    def forward(self, {input_signature}):
+        inputs = {input_dict}
+        results = self.inference_module(inputs)
+        #for name, preds in results.items():
+        #    for k, v in preds.items():
+        #        if v.dtype == torch.bool:
+        #            preds[k] = v.to(dtype=torch.int32)
```

Comment:
Is this comment here for a reason?

Reply:
Experimental code for binary types. Will remove for now.